code_conventions.md

```diff
@@ -1,108 +1,75 @@
+# @returns integer, the result of the something operation.
+def doSomething(someting: int) -> int
+  return something
+```
+``` python
+foo = 20
+foo += 20  # Add 20 to foo
+```
+
+## Logging
+As a rule of thumb, logging is used to aid developers with debugging purposes, both while developing new features and fixing bugs "in the office", as well as resolving issues with software that is out in the field.
+
+In order to provide some control and filtering to this logging, we've defined the following severity levels;
+* **DEBUG**: The most verbose logging. This is intended for logging data that is useful to debug the code. It can be used to indicate state changes.
+* **INFO**: Logging information that can be seen as feedback to a user on his/her actions, but without the information actually ending up with the user. For instance; Starting a print, print completed, etc.
+* **WARNING**: Warning message are an indication that something that should have happened is not entirely the case. This is different from ERROR in such that it's not yet guaranteed to cause the code to cease working (for instance, because there is a fallback mechanism)
+* **ERROR**: When piece of code cannot continue at all, but it doesn't  necessarily mean that the application has to stop working.
+* **CRITICAL**: Critical indicates that there is a situation that cannot be overcome without external influence. This implies that the application is very likely to stop working at all.
 
 Indenting / trailing whitespaces
 -----
-* Never use TABs
-* Indenting is always 4 spaces
-* No trailing whitespaces
+* You MUST NOT use TABs for indentation.
+* Indenting MUST be done with 4 spaces per indentation level
+* There MUST NOT be trailing whitespaces.
 
-Make sure that all editors used enforce these settings for the lines edited  changed (untouched code stays the same, unless a complete refactoring is going to happen)
+Make sure that all editors used enforce these settings, but only for the lines edited to ensure that untouched code remains the same.
 
 Localization *** TODO ***
 ----
-I8N strings?
-embedded gelijk aan uranium
-context marker + vertaling
-use named arguments - this will help in the long run!
-examples
+Although there is no full guideline for localization, the following things have been agreed on;
+* Localization MUST be done by using .po files.
+* You SHOULD use context hints
 
 Code blocks (not Code::Blocks :))
 -----
-* Always use a code block if possible and allowed in the language construction
-* Code blocks always start on a new line
-* The opening and closing code block delimiters should always be on a separate line on the same indentation level as the keywords (e.g. `if`, `while`, `else`).
-* Any code within the a code block must be indented to one indentation level further. Indentation levels differ by 4 spaces.
-
-Some examples of the previous rules:
-
-Bad code
-~~~~~~~~~~~~~~~{.cpp}
+* Codeblocks SHOULD be used if possible and allowed in the language construction
```

Comment:
Uhm, this can be interpreted in a lot of different ways. Reading it literally, I would write the following code:

```cpp
void doSomething(int value)
{
    int result;
    {
        for(int i = 0; i < value; ++i)
        {
             int temp = 0;
             {
                 temp = i * i;
                 {
                     temp += i;
                     {
                         result += temp;
                     }
                 }
             }
        }
    }
}

```

Or something similarly convoluted. This really needs more qualification.